src/ruby/puppet-server-lib/puppet/server/oids.rb

```diff
@@ -0,0 +1,31 @@
+module Puppet::Server::Oids
+  def self.load_custom_oid_file(custom_oid_file, map_key='oid_mapping')
+    if File.exists?(custom_oid_file) && File.readable?(custom_oid_file)
```

Comment:
I think it would be good to add a comment in here somewhere which points to the location of the file in the Ruby Puppet repo that this implementation was copied from.